src/main/scala/mesosphere/marathon/raml/PodStatusConversion.scala

```diff
@@ -249,36 +247,39 @@ trait PodStatusConversion {
     }.getOrElse(List.empty[ContainerEndpointStatus])
 
   def podInstanceState(
-    instanceCondition: core.condition.Condition,
+    instance: core.instance.Instance,
     containerStatus: Seq[ContainerStatus]): (PodInstanceState, Option[String]) = {
 
-    instanceCondition match {
-      case condition.Condition.Scheduled |
-        condition.Condition.Provisioned |
-        condition.Condition.Reserved =>
-        PodInstanceState.Pending -> None
-      case condition.Condition.Staging |
-        condition.Condition.Starting =>
-        PodInstanceState.Staging -> None
-      case condition.Condition.Error |
-        condition.Condition.Failed |
-        condition.Condition.Finished |
-        condition.Condition.Killed |
-        condition.Condition.Gone |
-        condition.Condition.Dropped |
-        condition.Condition.Unknown |
-        condition.Condition.Killing =>
-        PodInstanceState.Terminal -> None
-      case condition.Condition.Unreachable |
-        condition.Condition.UnreachableInactive =>
-        PodInstanceState.Degraded -> Some(MSG_INSTANCE_UNREACHABLE)
-      case condition.Condition.Running =>
-        if (containerStatus.exists(_.conditions.exists { cond =>
-          cond.name == STATUS_CONDITION_HEALTHY && cond.value == "false"
-        }))
-          PodInstanceState.Degraded -> Some(MSG_INSTANCE_UNHEALTHY_CONTAINERS)
-        else
-          PodInstanceState.Stable -> None
+    if (instance.isScheduled) {
+      PodInstanceState.Pending -> None
+    } else {
```

Comment:
I don't quite understand this change. We are checking for the scheduled condition below so why the extra if clause?

Reply:
look at how was `Reserved` used in the previous diff. For reserved, we showed state as pending. `Reserved` condition is dropped now. When thinking in general, I believe that`Pending` is the correct `PodInstanceState` for instances that are SCHEDULED - meaning `isScheduled` is true

Reply:
Ah, that relates to #6616 again where `condition = Scheduled` is not enough but one has to check `isScheduled`. Got it. Thanks for clarifying.